scylla-server/src/db_handler.rs

```diff
@@ -63,10 +63,14 @@ impl DbHandler {
                         info!("{} batches remaining!", batch_queue.len()+1);
                         // do not spawn new tasks in this mode, see below comment for chunk_size math
                         let chunk_size = final_msgs.len() / ((final_msgs.len() / 16380) + 1);
+                        if chunk_size == 0 {
+                            warn!("Could not insert {} messages, chunk size zero!", final_msgs.len());
```

Comment:
Could these warning messages (the one on line 87 as well) have different name that differentiate them? Unless final_msgs is literally just the last msgs and msgs are just what we use normal.

Other than that lgtm, cool logic with the Range map!

Reply:
Ya it's just a seperate section of code that runs on shutdown, does the same thing.  Only difference it does not send the db to background.